GpsDataCapturer/mobile/src/main/java/com/google/gpsdatacapturer/MobileGpsMainActivity.java

```diff
@@ -196,6 +208,29 @@ public void onServiceConnected(ComponentName name, IBinder service) {
             //get the gpsDataCaptureService
             gpsDataCaptureService =
                     ((GpsDataCaptureService.GpsDataCaptureBinder) service).getService();
+
+            //start capture via intent
+            if(adbIntent != null) {
+                //Extra the location api type
+                if(adbIntent.getExtras() != null){
+                    Log.d(TAG, "Intent extra is " + adbIntent.getExtras().toString());
+                    boolean type_from_intent = adbIntent.getBooleanExtra("fused_location_type", false);
```

Comment:
this extra would be present only for action com.google.gpsdatacapturer.START_CAPTURE
Also the user may forget to add it in the command line. Please make sure to add check for missing extra

Reply:
In current case, the default value is false (locationApiType = LocationManager) if user forgets adding the extra. The user only need to set it be true when they hope to use FusedLocationProvider. Does it make sense?

Reply:
You should check intent.hasExtra before using getExtras()